backend/backend/Controllers/UsersController.cs

```diff
@@ -191,36 +191,71 @@ public ActionResult ResetPassword(string newPassword)
+            {
+                return BadRequest();
+            }
             
             user.SetNewPassword(newPassword);
 
             _db.SaveChanges();
 
             return Ok();
         } 
-        [HttpPut, Route("update-credentials")]
-        public ActionResult UpdateCredentials([FromBody] ChangeCredentialsRequest request)
+        [HttpPatch, Route("{id}")]
+        public ActionResult UpdateUser([FromBody] ChangeCredentialsRequest request, [FromRoute] Guid id)
         {
-            if (String.IsNullOrWhiteSpace(request.Email) || !RegexValidation.IsEmailValid(request.Email)) return BadRequest();
-            if (String.IsNullOrWhiteSpace(request.OldPassword) || !RegexValidation.IsPasswordValid(request.OldPassword)) return BadRequest();
-            if (String.IsNullOrWhiteSpace(request.NewPassword) || !RegexValidation.IsPasswordValid(request.NewPassword)) return BadRequest();
-
-            if (request.OldPassword == request.NewPassword) return BadRequest();
-
             var userIdClaim = User.FindFirst(x => x.Type == ClaimTypes.NameIdentifier);
-            if (userIdClaim is null) return NotFound();
-            if (!Guid.TryParse(userIdClaim.Value, out var userId)) return NotFound();
+            if (userIdClaim is null) return NotFound("JWT doesn't correspond to any user");
+            if (!Guid.TryParse(userIdClaim.Value, out var userId)) return NotFound("JWT token is malformed");
+            if (id != userId) return BadRequest("Something wrong has happened");
             var user = _db.Users.FirstOrDefault(x => x.Id == userId);
-            if (user is null) return NotFound();
+            
+            if (user is null) return NotFound("JWT doesn't correspond to any user");
 
-            if (!Hasher.CheckPlaintextAgainstHash(request.OldPassword,user.Password,user.Salt)) return Unauthorized();
-            if (user.Email != request.Email && _db.Users.FirstOrDefault(u => u.Email == request.Email) != null) return Conflict();
+            if (request.Email != user.Email && _db.Users.FirstOrDefault(u => u.Email == request.Email) != null)
+                return Conflict("User with this email already exists");
 
-            user.Email = request.Email;
-            user.SetNewPassword(request.NewPassword);
+            if (string.IsNullOrWhiteSpace(request.Email) && string.IsNullOrWhiteSpace(request.OldPassword)
+                                                         && string.IsNullOrWhiteSpace(request.NewPassword))
+            {
+                return BadRequest("Request is empty");
+            }
 
-            _db.SaveChanges();
-            return Ok();
+
+            if (!string.IsNullOrWhiteSpace(request.Email)) user.Email = request.Email;
+            
+            if ((string.IsNullOrWhiteSpace(request.OldPassword) && !string.IsNullOrWhiteSpace(request.NewPassword)) 
+                || !string.IsNullOrWhiteSpace(request.OldPassword) && string.IsNullOrWhiteSpace(request.NewPassword))
```

Comment:
```suggestion
            if (string.IsNullOrWhiteSpace(request.OldPassword) || string.IsNullOrWhiteSpace(request.NewPassword))
```
Wouldn't this do the same?

Reply:
This here is exactly about the case where one of the fields is empty and the other one ain't. That means that the request is malformed and we should throw it out

In your case, there is a possibility that both of them are null, and in that case it doesn't necessarily mean that the request is bad

Reply:
But haven't we already covered email case above? Then my suggestion would be enough to cover the password cases. I mean to update password we need both of them: old and new, so wouldn't this suggested check be enough? I might be confused here, maybe you are right ✌🏻

Reply:
Wait since this endpoint is used for both Aurimas' profile page and yours Reset, right? So that means we need to only check either if both old and new is provided, or only new?

Reply:
Yes, I thought about that, but I'm saying that if one of them IS provided, then the request is malformed and we should discard it altogether, without updating email or w/e

No, different endpoints are used. It's kinda weird, I know, but reset password needs JWT Reset token, while this one doesn't need it.